### ISSUE TITLE
error while updating game

### ISSUE THREAD

**Issue by shalkya** [FAForever/downlords-faf-client#711]
i'm not sure if this is relevant, and if this is the appropriate place to post this. Pls tell me otherwise.
https://gist.github.com/shalkya/cdb9ad08b271e3b890929f8afc1dac6f
https://gist.github.com/shalkya/5cfe0b0d437d0dd787b487cfbd8ba93c

```
java.util.concurrent.CompletionException: com.github.nocatch.NoCatchException: java.nio.file.FileSystemException: C:\Program Files (x86)\THQ\Gas Powered Games\Supreme Commander - Forged Alliance\bin\BugSplat.dll -> C:\ProgramData\FAForever\bin\BugSplat.dll: Le processus ne peut pas accéder au fichier car ce fichier est utilisé par un autre processus.

	at java.util.concurrent.CompletableFuture.encodeRelay(Unknown Source)
	at java.util.concurrent.CompletableFuture.completeRelay(Unknown Source)
	at java.util.concurrent.CompletableFuture.uniRelay(Unknown Source)
	at java.util.concurrent.CompletableFuture$UniRelay.tryFire(Unknown Source)
	at java.util.concurrent.CompletableFuture.postComplete(Unknown Source)
	at java.util.concurrent.CompletableFuture.completeExceptionally(Unknown Source)
	at com.faforever.client.task.CompletableTask.lambda$new$1(CompletableTask.java:17)
	at com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:86)
	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:238)
	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:54)
	at javafx.event.Event.fireEvent(Event.java:198)
	at javafx.concurrent.EventHelper.fireEvent(EventHelper.java:219)
	at javafx.concurrent.Task.fireEvent(Task.java:1356)
	at javafx.concurrent.Task.setState(Task.java:707)
	at javafx.concurrent.Task$TaskCallable.lambda$call$502(Task.java:1453)
	at com.sun.javafx.application.PlatformImpl.lambda$null$172(PlatformImpl.java:295)
	at java.security.AccessController.doPrivileged(Native Method)
	at com.sun.javafx.application.PlatformImpl.lambda$runLater$173(PlatformImpl.java:294)
	at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:95)
	at com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
	at com.sun.glass.ui.win.WinApplication.lambda$null$147(WinApplication.java:177)
	at java.lang.Thread.run(Unknown Source)
Caused by: com.github.nocatch.NoCatchException: java.nio.file.FileSystemException: C:\Program Files (x86)\THQ\Gas Powered Games\Supreme Commander - Forged Alliance\bin\BugSplat.dll -> C:\ProgramData\FAForever\bin\BugSplat.dll: Le processus ne peut pas accéder au fichier car ce fichier est utilisé par un autre processus.

	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source)
	at java.lang.reflect.Constructor.newInstance(Unknown Source)
	at com.github.nocatch.NoCatch.wrapException(NoCatch.java:61)
	at com.github.nocatch.NoCatch.noCatch(NoCatch.java:104)
	at com.github.nocatch.NoCatch.noCatch(NoCatch.java:82)
	at com.faforever.client.patch.GameBinariesUpdateTaskImpl.lambda$copyGameFilesToFafBinDirectory$4(GameBinariesUpdateTaskImpl.java:130)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(Unknown Source)
	at java.util.stream.ReferencePipeline$2$1.accept(Unknown Source)
	at java.util.Iterator.forEachRemaining(Unknown Source)
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Unknown Source)
	at java.util.stream.AbstractPipeline.copyInto(Unknown Source)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(Unknown Source)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(Unknown Source)
	at java.util.stream.AbstractPipeline.evaluate(Unknown Source)
	at java.util.stream.ReferencePipeline.forEach(Unknown Source)
	at com.faforever.client.patch.GameBinariesUpdateTaskImpl.copyGameFilesToFafBinDirectory(GameBinariesUpdateTaskImpl.java:125)
	at com.faforever.client.patch.GameBinariesUpdateTaskImpl.call(GameBinariesUpdateTaskImpl.java:86)
	at com.faforever.client.patch.GameBinariesUpdateTaskImpl.call(GameBinariesUpdateTaskImpl.java:39)
	at javafx.concurrent.Task$TaskCallable.call(Task.java:1423)
	at java.util.concurrent.FutureTask.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	... 1 more
Caused by: java.nio.file.FileSystemException: C:\Program Files (x86)\THQ\Gas Powered Games\Supreme Commander - Forged Alliance\bin\BugSplat.dll -> C:\ProgramData\FAForever\bin\BugSplat.dll: Le processus ne peut pas accéder au fichier car ce fichier est utilisé par un autre processus.

	at sun.nio.fs.WindowsException.translateToIOException(Unknown Source)
	at sun.nio.fs.WindowsException.rethrowAsIOException(Unknown Source)
	at sun.nio.fs.WindowsFileCopy.copy(Unknown Source)
	at sun.nio.fs.WindowsFileSystemProvider.copy(Unknown Source)
	at java.nio.file.Files.copy(Unknown Source)
	at com.faforever.client.patch.GameBinariesUpdateTaskImpl.lambda$null$2(GameBinariesUpdateTaskImpl.java:130)
	at com.github.nocatch.NoCatch.noCatch(NoCatch.java:100)
	... 20 more
```

**Comment by micheljung** [comment]
Hi, it looks like some other process is messing with the files. If I had to guess I'd suspect an antivirus? Can you try disabling it or add an exception to:

`C:\ProgramData\FAForever`
`C:\Users\PierreYves\Documents\My Games\Gas Powered Games\Supreme Commander Forged Alliance`

**Comment by shalkya** [comment]
(keyser here)
I don't have antivirus but I removed the read only parameter from C:\ProgramData\FAForever\gamedata and was able to play on normal client yesterday.
Still wasn't working on your client, so i did the same for C:\ProgramData\FAForever (as you said) and seems to be working for your client now.
So I will close this topic. Not sure how this happened but can be useful for other people that encounter it maybe. (I downloaded old patch to watch replay, so that's maybe due to that idk)
